src/lib.rs

```diff
@@ -450,6 +450,45 @@ impl From<io::Error> for GooseError {
     }
 }
 
+/// Statistics collected during a Goose load test.
+///
+/// # Example
+/// ```rust,no_run
+///     use goose::prelude::*;
+///
+/// fn main() -> Result<(), GooseError> {
+///     let stats = GooseAttack::initialize()?
+///         .register_taskset(taskset!("ExampleUsers")
+///             .register_task(task!(example_task))
+///         )
+///         .execute()?
+///         .get_stats();
+///
+///     // Do something with stats ...
+///     println!("{:#?}", stats);
+///
+///     Ok(())
+/// }
+///
+/// async fn example_task(user: &GooseUser) -> GooseTaskResult {
+///     let _goose = user.get("/").await?;
+///
+///     Ok(())
+/// }
+/// ```
+#[derive(Clone, Debug)]
+pub struct GooseStats {
+    /// A hash of the load test, useful to verify if different statistics are from
+    /// the same load test.
+    pub hash: u64,
+    /// How many seconds the load test ran.
+    pub run_time: usize,
```

Comment:
The GooseAttack struct has `run_time` also,  confusing to have both attributes with the same name or useful?

Also, in my case at least I want to know the time the test will run before running it so the `run_time` in GooseAttack, not sure if it makes sense to make fields on that struct pub also.

Reply:
My original intent was that these are the same name because they are the same value. But if you ctrl-c a test before it finishes, they would not be the same value. So in `GooseStats` I've now renamed it to `duration`.

You would have to set `run_time` in your code, as otherwise Goose runs forever. So it seems you already have access to this information?

Reply:
Yes, but I think then the parse_time function in utils would need to be public? I want time as usize, already parsed from command-line.

Reply:
I believe you'd do it the same as the integration tests? For example: https://github.com/tag1consulting/goose/blob/master/tests/common.rs#L12 <-- this sets it to 1 second, but that could be set to any desired number of seconds. Inside `GooseStats` the `duration` is returned as a `usize` measured in seconds.

Reply:
I want to be able to pass "3m" or "4h" though the command line as Goose does but work later with an usize.
The reason I want the usize is because I am [displaying a progress bar](https://github.com/JordiPolo/minos/blob/master/src/command_performance.rs#L38-L57) while the user is waiting for the final stats. Maybe I should implement this inside Goose directly, are you interested?

Reply:
A progress bar in goose would be very welcome.